src/NetPanzer/Units/UnitOpcodes.hpp

```diff
@@ -40,12 +40,7 @@ struct UnitOpcodeStruct
 {
 public:
     Uint8 opcode;
-private:
-    Uint16 unit_index;
```

Comment:
this doesn't effect the protocol over the wire does it? I am not sure if we just send a pointer over the wire yet or if we actually do it the "right way". See if you can still connect to a server and move your units :D

Reply:
@andy5995 can you still connect to an existing server and play with this change?

Reply:
When I remove those two variables that the compiler reports as "unused", I crash when trying to connect to a server. On one try, I connected, but didn't have any tanks that I could find.

Reply:
To be fair - I had a friend last night connect and spawn with no tanks. So that might just be a bug.
Have you seen that @ph2708 ?

Reply:
> To be fair - I had a friend last night connect and spawn with no tanks. So that might just be a bug. Have you seen that @ph2708 ?

Maybe the universe is trying to tell us something. Last night, I wanted to implement "observer mode". Joining with no tanks would be the first step. lol

Reply:
Yes, many games nowadays provide a Spectator mode.

I think for that, what we should have is a "Limbo" menu. You open this to change teams or join spectators, example: https://et.clan-etc.de/wolfguide/images/limbo_menu.jpg

When we support objectives etc this info can be shown on the larger map view, too.

I think it would be good to go implement this, however new UIs should not be using hard-coded offsets and things, we should have a Table component for example that can do layouts.